sw/airborne/modules/datalink/mavlink.c

```diff
@@ -107,22 +138,30 @@ void mavlink_init(void)
  */
 void mavlink_periodic(void)
 {
-  RunOnceEvery(2, mavlink_send_heartbeat());
-  RunOnceEvery(5, mavlink_send_sys_status());
-  RunOnceEvery(20, mavlink_send_system_time());
-  RunOnceEvery(10, mavlink_send_attitude());
-  RunOnceEvery(5, mavlink_send_attitude_quaternion());
-  RunOnceEvery(5, mavlink_send_params());
-  RunOnceEvery(4, mavlink_send_local_position_ned());
-  RunOnceEvery(5, mavlink_send_global_position_int());
-  RunOnceEvery(6, mavlink_send_gps_raw_int());
-  RunOnceEvery(5, mavlink_send_attitude_quaternion());
-  RunOnceEvery(5, mavlink_send_rc_channels());
-  RunOnceEvery(21, mavlink_send_battery_status());
-  RunOnceEvery(32, mavlink_send_autopilot_version());
-  RunOnceEvery(33, mavlink_send_gps_global_origin());
-  RunOnceEvery(10, mavlink_send_gps_status());
-  RunOnceEvery(10, mavlink_send_vfr_hud());
+#if PERIODIC_TELEMETRY && defined TELEMETRY_MAVLINK_NB_MSG
+  // send periodic mavlink messages as defined in the Mavlink process of the telemetry xml file
+  // transport and device not used here yet...
+  periodic_telemetry_send_Mavlink(&mavlink_telemetry, NULL, NULL);
+#else
```

Comment:
I think this RunOnceEvery was needed before because there was no other option, but now that you have the telemetry file which is much more flexible, I would argue that this RunOnceEvery can be completely removed.